core/src/test/java/org/apache/iceberg/TestMetadataUpdateParser.java

```diff
@@ -1030,49 +1031,45 @@ public void assertEquals(
             (MetadataUpdate.SetCurrentViewVersion) actualUpdate);
         break;
       default:
-        Assert.fail("Unrecognized metadata update action: " + action);
+        fail("Unrecognized metadata update action: " + action);
     }
   }
 
   private static void assertEqualsAssignUUID(
       MetadataUpdate.AssignUUID expected, MetadataUpdate.AssignUUID actual) {
-    Assert.assertEquals("UUIDs should be equal", expected.uuid(), actual.uuid());
+    assertThat(actual.uuid()).isEqualTo(expected.uuid());
   }
 
   private static void assertEqualsUpgradeFormatVersion(
       MetadataUpdate.UpgradeFormatVersion expected, MetadataUpdate.UpgradeFormatVersion actual) {
-    Assert.assertEquals(
-        "Format version should be equal", expected.formatVersion(), actual.formatVersion());
+    assertThat(actual.formatVersion()).isEqualTo(expected.formatVersion());
   }
 
   private static void assertEqualsAddSchema(
       MetadataUpdate.AddSchema expected, MetadataUpdate.AddSchema actual) {
-    Assert.assertTrue("Schemas should be the same", expected.schema().sameSchema(actual.schema()));
-    Assert.assertEquals(
-        "Last column id should be equal", expected.lastColumnId(), actual.lastColumnId());
+    assertThat(actual.schema().sameSchema(expected.schema())).isTrue();
```

Comment:
```suggestion
    assertThat(actual.schema().asStruct()).isEqualTo(expected.schema());
```

this will show the schemas if the assertion fails